File: java/src/main/java/frc/engineer/dto/DtoModels.java
package frc.engineer.dto;

import java.util.ArrayList;
import java.util.List;

public final class DtoModels {
    private DtoModels() {
    }

    public static final class PitwallState {
        public Integer focusedCarIndex;
        public List<String> notes = new ArrayList<>();
        public List<String> warnings = new ArrayList<>();
    }

    public static final class CarDamage {
        public Integer frontLeftWing;
        public Integer frontRightWing;
        public Integer rearWing;
        public Integer floor;
        public Integer diffuser;
        public Integer sidepod;
        public Integer gearbox;
        public Integer engine;
    }

    public static final class TyreSetBrief {
        public Integer index;
        public String compound;
        public Integer ageLaps;
        public Integer wear;
        public Boolean available;
    }

    public static final class StintInfoDTO {
        public Integer startLap;
        public Integer endLap;
        public String compound;
        public Integer totalLaps;
    }

    public static final class DynamicsLapDTO {
        public Integer lapNumber;
        public Float fuelInTank;
        public Integer ersPct;
        public Float tireWearAvg;
    }
}

File: kotlin/src/main/kotlin/frc/engineer/dto/DtoModels.kt
package frc.engineer.dto

data class PitwallState(
    val focusedCarIndex: Int? = null,
    val notes: List<String> = emptyList(),
    val warnings: List<String> = emptyList(),
)

data class CarDamage(
    val frontLeftWing: Int? = null,
    val frontRightWing: Int? = null,
    val rearWing: Int? = null,
    val floor: Int? = null,
    val diffuser: Int? = null,
    val sidepod: Int? = null,
    val gearbox: Int? = null,
    val engine: Int? = null,
)

data class TyreSetBrief(
    val index: Int? = null,
    val compound: String? = null,
    val ageLaps: Int? = null,
    val wear: Int? = null,
    val available: Boolean? = null,
)

data class StintInfoDTO(
    val startLap: Int? = null,
    val endLap: Int? = null,
    val compound: String? = null,
    val totalLaps: Int? = null,
)

data class DynamicsLapDTO(
    val lapNumber: Int? = null,
    val fuelInTank: Float? = null,
    val ersPct: Int? = null,
    val tireWearAvg: Float? = null,
)
